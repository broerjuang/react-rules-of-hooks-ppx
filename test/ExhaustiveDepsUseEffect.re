[@react.component]
let make = (~randomProp as _: string) => {
  let (show, _setShow) = React.useState(() => "sTatE");

  React.useEffect1(
    () => {
      Js.log(randomProp);
      None;
    },
    [|show|],
  );

  <div />;
};
