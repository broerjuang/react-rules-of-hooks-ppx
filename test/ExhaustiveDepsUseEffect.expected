[1mFile "ExhaustiveDepsUseEffect.ml", line 3, characters 2-63[0m:
3 |   React.useEffect1 (fun () -> Js.log randomProp; None) [|show|];
      [1;31m^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^[0m
[1;31mError[0m: ExhaustiveDeps: Missing 'randomProp' in the dependency array
